File: Arrays/Medium/Dutch_National_Flag.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    void sortZeroOneTwo(vector<int>& nums) {
        int low = 0;
        int mid = 0;
        int high = (int)nums.size() - 1;
        while (mid <= high) {
            if (nums[mid] == 0) {
                swap(nums[low], nums[mid]);
                low++;
                mid++;
            } else if (nums[mid] == 1) {
                mid++;
            } else { // nums[mid] == 2
                swap(nums[mid], nums[high]);
                high--;
            }
        }
    }
};

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin >> n;
        vector<int> arr(n);
        for (auto& a : arr){
            cin >> a;
        }
        Solution sol;
        sol.sortZeroOneTwo(arr);

        for(auto& a: arr){
            cout << a;
        }
        cout<<endl;
    }
}

File: Arrays/Medium/Dutch_National_Flag.py
class Solution:
    def sortIt(self,nums):
        low = 0
        mid = 0
        size = len(nums)
        high = size-1
        for i in range(size):
            if(nums[mid] == 0):
                nums[mid], nums[low] = nums[low], nums[mid]
                low +=1
                mid +=1
            elif(nums[mid] == 1):
                mid +=1
            else:
                nums[mid], nums[high] = nums[high], nums[mid]
                high -=1
        
        return nums

sol = Solution()
t = int(input())
nums = []
while(t):
    nums.append(int(input()))
    t-=1

print(sol.sortIt(nums))